inst/tests/tests.Rraw

```diff
@@ -17295,3 +17295,7 @@ test(2165.3, X[Y],                          data.table(A=2:3, B=2:3, i.A=2:1, ke
 test(2165.4, X[Y, on=.(A)],                 data.table(A=2:1, B=c(2L,NA), i.B=2:3))      # no key
 test(2165.5, X[Y, on=.(A), x.B, by=.EACHI], data.table(A=2:1, x.B=c(2L,NA)))             # no key
 
+# `keyby`=TRUE/FALSE together with by=, #4307
+DT = data.table(a=2:1, b=3:2, d=4:3)
+test(2166.1, DT[, .SD, by="a"], DT[, .SD, by="a", keyby=FALSE])
+test(2166.2, DT[, .SD, keyby="a"], DT[, .SD, by="a", keyby=TRUE])
```

Comment:
`y` should be a fixed known value such as returned by a `data.table()` call please.  So that these tests don't falsely pass in future after a change that makes both `x` and `y` wrong but still matching each other.